3 Module Development Exercise - Recursive Unhider/recursiveUnhider-rt/src/com/tridiumuniversity/recursiveUnhider/BUnhider.java

```diff
@@ -0,0 +1,73 @@
+package com.tridiumuniversity.recursiveUnhider;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.Action;
+import javax.baja.sys.BComponent;
+import javax.baja.sys.Context;
+import javax.baja.sys.Flags;
+import javax.baja.sys.Slot;
+import javax.baja.sys.Sys;
+import javax.baja.sys.Type;
+import java.util.logging.Logger;
+
+@NiagaraType
+@NiagaraAction(
+        name = "unhide",
+        flags = Flags.ASYNC
+)
+public class BUnhider extends BComponent {
+//region /*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+//@formatter:off
+/*@ $com.tridiumuniversity.recursiveUnhider.BUnhider(3895808588)1.0$ @*/
+/* Generated Thu Sep 26 19:42:08 CEST 2024 by Slot-o-Matic (c) Tridium, Inc. 2012-2024 */
+
+  //region Action "unhide"
+
+  /**
+   * Slot for the {@code unhide} action.
+   * @see #unhide()
+   */
+  public static final Action unhide = newAction(Flags.ASYNC, null);
+
+  /**
+   * Invoke the {@code unhide} action.
+   * @see #unhide
+   */
+  public void unhide() { invoke(unhide, null, null); }
+
+  //endregion Action "unhide"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BUnhider.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+    public void doUnhide (Context context) {
+        removeHiddenFlagFromChildren((BComponent) getParent(), context);
+    }
+    private void removeHiddenFlagFromChildren(BComponent component, Context context) {
+      removeHiddenFlagFromSlots(component, context);
+      if (component.getChildComponents() == null) {
```

Comment:
The return value from `getChildComponents()` won't ever be null. It's a `BComponent[]`, so if there are no child components, then it will be an empty array.